.travis.yml

```diff
@@ -1,32 +1,64 @@
-language: rust
-dist: trusty
-sudo: required
+language: python
+sudo: false
+
+python: "3.4"
 
 env:
+  global:
     - PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$HOME/install/syslog-ng/lib/pkgconfig
+    - CMAKE_C_FLAGS=-std=gnu99
+    - LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$HOME/install/syslog-ng/lib"
+    - PATH="$HOME/.local/bin:$PATH"
+  matrix:
+    - RUST_CHANNEL=stable TEST_MODE=everything
+    - RUST_CHANNEL=beta TEST_MODE=everything
+    - RUST_CHANNEL=nightly TEST_MODE=everything TRAVIS_CARGO_NIGHTLY_FEATURE=nightly
 
-# run builds for all the trains (and more)
 matrix:
   include:
-    - rust: stable
-
-before_script:
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq autoconf-archive bison cmake docbook-xsl flex libcap-dev libesmtp-dev libevtlog-dev libgeoip-dev libglib2.0-dev libhiredis-dev libjson0-dev libnet1-dev libpcre3-dev libssl-dev libwrap0-dev openjdk-7-jdk pkg-config python3-all-dev libpython3-dev libpython3-all-dev time xsltproc
-  - cd ..
-  - git clone https://github.com/balabit/syslog-ng.git
-  - cd syslog-ng
-  - ./autogen.sh
-  - ./configure --with-ivykis=internal --prefix=$HOME/install/syslog-ng --disable-python --disable-java --disable-sql --disable-amqp
-  - make -j install
-  - cd ../syslog-ng-rust-modules
-  - mkdir build
-  - cd build
-  - >
-    cmake -DCMAKE_INSTALL_PREFIX=$HOME/install/syslog-ng
-    -DENABLE_REGEX_PARSER=ON
-    -DENABLE_PYTHON_PARSER=ON
-    -DENABLE_ACTIONDB_PARSER=ON
-    -DENABLE_CORRELATION_PARSER=ON ..
+    - env: RUST_CHANNEL=stable TEST_MODE=cmake
+
+git:
+  submodules: false
+
+dist: trusty
+cache:
+  - cargo
+  - ccache
```

Comment:
I haven't tried this combination yet. The documentation says that both `language: c` and `cache: ccache` is needed to enable automatic `ccache`.
https://docs.travis-ci.com/user/caching/#ccache-cache

To enable `ccache` manually:
* `cache: $HOME/.ccache`
* ...
*  `- export PATH="/usr/lib/ccache:$PATH"`
* `- echo "max_size = 150M" > "$HOME/.ccache/ccache.conf"`

For debugging whether this works in practice, I usually add this to the `before_script` and `after_script` respectively:
*  `- ccache -z`
* `- ccache -s`

I find that as `ccache` can interfere with `clang`, it is usually disabled by default. I experimented with force enabling it, but didn't finish. I tried a few variations, like:

* ` printf "run_second_cpp = true\nprefix_command = -Qunused-arguments\n" >> ~/.ccache/ccache.conf;`
* ` if ! [ -f $HOME/.local/bin/clang ]; then ln -s $(which ccache) $HOME/.local/bin/clang; fi`
* `printf '#!/bin/sh\nCCACHE_CPP2=yes exec ccache clang "$@"\n' > $HOME/.local/bin/clang; chmod +x $HOME/.local/bin/clang`
* `echo 'CCACHE_CPP2=yes exec ccache clang -Qunused-arguments $(test -t 2 && echo -fcolor-diagnostics) "$@"' > $HOME/.local/bin/clang; chmod +x $HOME/.local/bin/clang`
* ...